src/django/api/serializers/user_api_info_serializer.py

```diff
@@ -0,0 +1,49 @@
+
+from rest_framework.exceptions import ValidationError
+from api.models import (RequestLog, ApiLimit)
+from api.models.contributor.contributor import Contributor
+
+
+class UserApiInfoSerializer:
+    uid: int
+    contributor_id: int
+
+    def __init__(self, uid):
+        self.uid = uid
+
+        try:
+            contributor = Contributor.objects.get(admin_id=self.uid)
+            self.contributor_id = contributor.id
+        except Contributor.DoesNotExist:
+            raise ValidationError((
+                    f"User with id = {self.uid} doesn't has a contributor"
+            ))
+
+    def get_api_call_limit(self):
```

Comment:
You had better define the properties of the serializer and then use the serializer in the appropriate way.
```
data = MyCustomSerializer(arg1, arg2).data
api_call_info_data = {
            "apiCallAllowance": data['api_call_limit'],
            "currentCallCount": data['current_usage'],
            "renewalPeriod": data['renewal_period'],
        }
```

Reply:
rewritten